src/node_i18n.cc

```diff
@@ -185,13 +219,379 @@ static void ToASCII(const FunctionCallbackInfo<Value>& args) {
                           len).ToLocalChecked());
 }
 
+// Get's the codepoint at a given offset for UTF-8 or UCS2
+// args[0] must be a buffer instance
+// args[1] must be a boolean, true = utf8, false = ucs2
+// args[2] must be the integer offset within the buffer to check
+void GetCodePointAt(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  SPREAD_ARG(args[0], ts_obj);
+
+  CHECK(args[1]->IsBoolean());
+  CHECK(args[2]->IsUint32());
+
+  bool utf8 = args[1]->BooleanValue();
+  uint32_t pos = args[2]->Uint32Value();
+  UChar32 codepoint = 0;
+
+  if (utf8) {
+    U8_GET_UNSAFE(ts_obj_data, pos, codepoint);
+  } else {
+    if (IsBigEndian()) {
+      const uint32_t len = ts_obj_length >> 1;
+      UChar* copy = static_cast<UChar*>(malloc(len));
```

Comment:
Ack… alternatively, maybe use `MaybeStackBuffer<UChar> copy(len);` here, too? (and basically ditto for all following `malloc()` calls). Also, this seems too short? Shouldn’t it be `malloc(len << 1)`?